src/components/Helpers.tsx

```diff
@@ -0,0 +1,22 @@
+import { Todo } from '../types/Todo';
+
+export const filterTodos
+= (initialTodos: Todo[], filter:string, query: string) => {
+  const filteredTodos = initialTodos.filter((todo: Todo) => {
+    const normalizedQuery = query.trim().toLowerCase();
+    const isIncludes = todo.title.toLowerCase().includes(normalizedQuery);
```

Comment:
You are creating normilizedQuery every iteration 

```suggestion
export const filterTodos
= (initialTodos: Todo[], filter:string, query: string) => {
  const normalizedQuery = query.trim().toLowerCase();
  
  const filteredTodos = initialTodos.filter((todo: Todo) => {
    const isIncludes = todo.title.toLowerCase().includes(normalizedQuery);
```